### PR TITLE
User CRUD

### PR DESCRIPTION
**Pull request by Erikad94**
Nous avons ajouté le CRUD user et les tests d'intégrations

### CODE REVIEW

File: backend/test/integration/user.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+      user.email = 'something@something.com';
+      user.hashPassword = 'password';
+        user.save(
+          function(err, res){
+            userId = res._id;
+            should.not.exist(err);
+            done();
+          });
+    });
+    after(function() {
```
Comment (MicLamb) [inline backend/test/integration/user.js]: Only a question here : what is this "after" used for? Can I use something like this in the seed file?

**Comment by MicLamb** [comment]
I know you know it but I just want to mentionned that in the server.js we could probably do a refactor to put the routes somewhere else. Also, which function is used for the update? The create function? 

Very complete tests. :+1:

File: backend/test/integration/user.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        .expect(400) //Status code
+        .end(function(err, res) {
+          should.not.exist(err);
+          should.exist(res);
+          res.body.should.have.property('message', 'User validation failed');
+          res.body.should.have.property('name', 'ValidationError');
+          done();
+      });
+    });
+     it('should not be able to patch the user with a wrong email', function(done) {
```
Comment (MicLamb) [inline backend/test/integration/user.js]: It is the test for the wrong id?

File: backend/test/integration/user.js [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+        .end(function(err, res) {
+          should.not.exist(err);
+          should.exist(res);
+          res.body.should.have.property('message', 'User not found');
+          done();
+      });
+    });
+  });
+
+  describe('Patch', function() {
```
Comment (MicLamb) [inline backend/test/integration/user.js]: Is it the test for the delete?

**Comment by bluespart** [comment]
Seems good :+1:

File: backend/test/integration/user.js (hunk 1 above)
Comment (Erikad94) [inline backend/test/integration/user.js]: After the tests do this, is how i understand it

File: backend/test/integration/user.js (hunk 2 above)
Comment (Erikad94) [inline backend/test/integration/user.js]: True true I'll change it.

**Comment by Baltazar510** [comment]
Everything seem good.

jue dit chill

**Comment by pellerinse2302** [comment]
:+1:

**Pull request merged**